### ISSUE TITLE
[Bug][zos_linefile] Does not behave like community version, you end up with a duplicate entry

### ISSUE THREAD

**Issue by ddimatos** [ansible-collections/ibm_zos_core#583]
### Bug description

When you run the module twice you end up with two entries of the same line using zos_blockinfile where with the community version you do not. zos_linefile should align to the community version. 

Using zos_blockinfile you end up with two entries:
```

---
- hosts: zvm
  gather_facts: false
  environment: "{{ environment_vars }}"
  collections:
    - ibm.ibm_zos_core

  tasks:
    # - name: Touch a file
    #   ansible.builtin.file:
    #     path: /tmp/ansible_file1
    #     state: touch

    - name: Linefile
      zos_lineinfile:
        src: /tmp/ansible_file1
        state: present
        line: "ABCD"

    - name: Linefile
      zos_lineinfile:
        src: /tmp/ansible_file1
        state: present
        line: "ABCD"
```

```
# cat ansible_file1
ABCD
ABCD
```

This community usage results in only one entry in the file  `/tmp/ansible_file1`, note that using the community version.
```
---
- hosts: zvm
  gather_facts: false
  environment: "{{ environment_vars }}"
  collections:
    - ibm.ibm_zos_core

  tasks:
    - name: Linefile
      lineinfile:
        path: /tmp/ansible_file1
        state: present
        line: "ABCD"

    - name: Linefile
      lineinfile:
        path: /tmp/ansible_file1
        state: present
        line: "ABCD"
```

```
# cat ansible_file1
ABCD
```

### Ansible version

```shell
2.9
```

### IBM z/OS Ansible core Version

v1.4.0-beta.2

### IBM ZOAU version

v1.1.1

### z/OS version

2.4

### Ansible module

zos_lineinfile

**Comment by fernandofloresg** [comment]
Hi @ddimatos I was able to reproduce this issue, current present() function leverages ZOAU datasets.lineinfile, I'm using ZOAU version 1.2.1 and still facing the same issue.

 `def present(src, line, regexp, ins_aft, ins_bef, encoding, first_match, backrefs):
    """Replace a line with the matching regex pattern
    Insert a line before/after the matching pattern
    Insert a line at BOF/EOF

    Arguments:
        src: {str} -- The z/OS USS file or data set to modify.
        line: {str} -- The line to insert/replace into the src.
        regexp: {str} -- The regular expression to look for in every line of the src.
            If regexp matches, ins_aft/ins_bef will be ignored.
        ins_aft: {str} -- Insert the line after matching '*regex*' pattern or EOF.
            choices:
                - EOF
                - '*regex*'
        ins_bef: {str} -- Insert the line before matching '*regex*' pattern or BOF.
            choices:
                - BOF
                - '*regex*'
        encoding: {str} -- Encoding of the src.
        first_match: {bool} -- Take the first matching regex pattern.
        backrefs: {bool} -- Back reference

    Returns:
        str -- Information in JSON format. keys:
            cmd: {str} -- dsed shell command
            found: {int} -- Number of matching regex pattern
            changed: {bool} -- Indicates if the source was modified.
    """
    return datasets.lineinfile(
        src,
        line,
        regex=regexp,
        ins_aft=ins_aft,
        ins_bef=ins_bef,
        encoding=encoding,
        first_match=first_match,
        backref=backrefs,
        state=True,
        debug=True,
    )
`

Simplest solution imo is raise this issue with ZOAU team, otherwise we would have to refactor from the ground up the present function to implement a workaround.

**Comment by ddimatos** [comment]
@fernandofloresg - I agree, I also recreated and compared to the community lineinefile. I looked over ZOAU source and ran some tests and determined their API is not honoring "state=True -> Insert or replace line" , their API is building this command `dmod -d -c IBM-1047 "$ a\ABCD" ./afile` which does an APPEND to EOF , rather maybe they should build this command `dmod "/ABCD/c\ABCD/$" ./afile`

ZOAU tracker 9645

**Comment by ddimatos** [comment]
I believe all we need for this item when the patch is released for ZOAU is to add a test case.

**Comment by storfibm** [comment]
This will be fixed in ZOAU 1.2.4 - Q2 release target

**Comment by AndreMarcel99** [comment]
Is fixed with ZOAU 1.2.4 
```
  tasks:
    - name: Touch a file
      ansible.builtin.file:
        path: /tmp/ansible_file1
        state: touch

    - name: Linefile
      zos_lineinfile:
        src: /tmp/ansible_file1
        state: present
        line: "ABCD"

    - name: Linefile
      zos_lineinfile:
        src: /tmp/ansible_file1
        state: present
        line: "ABCD"

    - name: Cat result
      ansible.builtin.shell:
        cmd: cat ansible_file1
        chdir: tmp/
      register: result

    - name: Print result
      debug:
        msg: "{{result}}"
```

Result:
```
ok: [zvm] => {
    "msg": {
        "changed": true,
        "cmd": "cat ansible_file1",
        "delta": "0:00:00.272112",
        "end": "2023-07-17 17:53:43.624869",
        "failed": false,
        "rc": 0,
        "start": "2023-07-17 17:53:43.352757",
        "stderr": "",
        "stderr_lines": [],
        "stdout": "ABCD",
        "stdout_lines": [
            "ABCD"
        ]
    }
}
```
